autotest/gcore/cog.py

```diff
@@ -258,6 +258,27 @@ def my_cbk(pct, _, arg):
     gdal.GetDriverByName('GTiff').Delete(filename)
     gdal.Unlink(directory)
 
+###############################################################################
+# Test creation of overviews with a different compression method
+
+def test_cog_creation_of_overviews_with_compression():
+    directory = '/vsimem/test_cog_creation_of_overviews_with_compression'
+    filename = directory + '/cog.tif'
+    src_ds = gdal.Translate('', 'data/byte.tif',
+                            options='-of MEM -outsize 2048 300')
+
+    with gdaltest.config_option('COMPRESS_OVERVIEW', 'JPEG'):
+        ds = gdal.GetDriverByName('COG').CreateCopy(filename, src_ds,
+                                                options = ['COMPRESS=LZW'])
+
+        assert ds.GetRasterBand(1).GetOverviewCount() == 2
+        assert ds.GetMetadata('IMAGE_STRUCTURE')['COMPRESSION'] == 'LZW'
```

Comment:
I am not sure how to get the compression metadata for a overview, I couldn't find it in the docs

Reply:
you can open ``GTIFF_DIR:x:foo.tif`` where x=1 for full resolution image, 2 for first overview level, etc.

Reply:
took me way to long to realise that `GTIFF_DIR:` is a magic command for `gdal.Open` :) 

These tests now assert the overview compression